src/disp/lv_disp.c

```diff
@@ -139,6 +139,11 @@ void lv_disp_remove(lv_disp_t * disp)
 
     lv_disp_send_event(disp, LV_EVENT_DELETE, NULL);
 
+    uint32_t i, event_count = lv_disp_get_event_count(disp);
+    for(i = 0; i < event_count; i++) {
+        lv_disp_remove_event(disp, i);
```

Comment:
I think it won't work well. If there are 5 events:
- i = 0, delete event0, 4 events remain (1, 2, 3, 4)
- i = 1, delete event2 (event1 is at index 0), 3 events remain (1, 3, 4)
- i = 2, delete event4,  2 events remain (1, 3)
- i = 3, no event at this index
- i = 4, no event at this index


```suggestion
        lv_disp_remove_event(disp, 0);  /*Aways delete the first event*/
```

or 
```suggestion
        while(lv_disp_remove_event(disp, 0));
```

Probably we need to the same with objects [here](https://github.com/lvgl/lvgl/blob/master/src/core/lv_obj_tree.c#L389-L391).

Reply:
You are right. I will update it later.

Reply:
Maybe we can access event_list directly, not by API.